packages/amplication-server/src/core/build/build.service.ts

```diff
@@ -484,7 +481,11 @@ export class BuildService {
               adminUIPath: resourceInfo.settings.adminUISettings.adminUIPath,
               serverPath: resourceInfo.settings.serverSettings.serverPath,
             },
-            pullRequestMode,
+            pullRequestMode:
```

Comment:
What is the value of `pullRequestMode` if `smartGitSyncEntitlement` returns `false` (meaning typeof boolean)?

Reply:
I think you can do something like that:
declare the pull request mode
```suggestion
          const pullRequestMode =
            (smartGitSyncEntitlement &&
              smartGitSyncEntitlement.hasAccess &&
              EnumPullRequestMode.Accumulative) ||
            EnumPullRequestMode.Basic; 
```
and then use it here